### PR TITLE
imagePullsecret check if it exists

### PR DESCRIPTION
**Pull request by rahug20**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

- Referencing `image_pull_secret` variable in docker secret file
- Adding a check to see if the docker secret already exists using helm lookup

2. What **changes to custom.yaml** are required?

<br/><br/>

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

I ran `make secrets` and secrets were created in the cluster

### CODE REVIEW

**Comment by moosecanswim** [comment]
The logic in the lookup is good but i second david's comment.

**Comment by rahug20** [comment]
I pushed an update, updating all occurrences of hard coded docker.secret value

File: docs/Getting Started.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ Observables can be enabled or disabled for each service.  You can enable observa
 
 ### Docker credentials
 
-Before installing the secrets chart, or your own series of secrets, a docker secret with docker credentials to pull the Grey Matter images must be created. By default, it is named `docker.secret`, but a secret with a different name can be used as long as `.Values.global.image_pull_secret` is set to the name of the secret in all of the Grey Matter charts.
+Before installing the secrets chart, or your own series of secrets, a docker secret with docker credentials to pull the Grey Matter images must be created. By default, it is named `{{ .Values.global.image_pull_secret }}`, but a secret with a different name can be used as long as `.Values.global.image_pull_secret` is set to the name of the secret in all of the Grey Matter charts.
```
Comment (dgoldstein1) [inline docs/Getting Started.md:111]: This is now out of date

File: docs/Getting Started.md [hunk 2]
```diff
@@ -136,7 +136,7 @@ To add another docker registry with credentials to the secrets, simply add anoth
     password:
 ```
 
-To install, run `make secrets`. The `docker.secret` will have creall registries in this file.  If you need credentials please contact [Grey Matter Support](https://support.deciphernow.com).
+To install, run `make secrets`. The `{{ .Values.global.image_pull_secret }}` will have creall registries in this file.  If you need credentials please contact [Grey Matter Support](https://support.deciphernow.com).
```
Comment (dgoldstein1) [inline docs/Getting Started.md:139]: Spelling

**Comment by rahug20** [comment]
updated docs

**Comment by dgoldstein1** [comment]
Thanks for updating the docs. Im testing this and getting error:

```
 make install
(cd spire && make clean-spire)
rm -f ./charts/*
(cd fabric && make clean-fabric)
rm -f ./charts/*
(cd data && make clean-data)
rm -f ./charts/*
(cd sense && make clean-sense)
rm -f ./charts/*
(cd spire && make package-spire)
rm -f ./charts/*
echo "target hit package-spire"
target hit package-spire
helm dep up .
Error: cannot load values.yaml: error converting YAML to JSON: yaml: invalid map key: map[interface {}]interface {}{".Values.global.image_pull_secret":interface {}(nil)}
make[1]: *** [package-spire] Error 1
make: *** [dev-dep] Error 2
```

In global.yaml I have:
```yaml
  image_pull_secret: docker.secret.test
```

**Comment by rahug20** [comment]
I had to push an update reverting my previous commit of updating all references of `docker.secret` variable value. The `docker.secret` value is referenced in all of the greymatter services values.yaml files as a default value which we need to have.
When we run `make install` the cluster uses the value from the main `global.yaml` file to get the value of the image_pull_secret.
Run `make install` and once the cluster deploys check the stateful set of any service and you will see the image_pull_secret value to be whatever you set the value of image_pull_secret in the main `global.yaml` file.

**Comment by rahug20** [comment]
The purpose of this ticket was to reference the `image_pull_secret` value in docker-secret.yaml file. Also, to check if the secret with the specified name exists or not.

**Pull request merged**